code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm

```diff
@@ -73,31 +72,13 @@
 
 	return FALSE
 
-/mob/living/simple_animal/hostile/asteroid/gutlunch/Destroy()
-	QDEL_NULL(udder)
-	return ..()
-
-/mob/living/simple_animal/hostile/asteroid/gutlunch/regenerate_icons()
+/mob/living/simple_animal/hostile/asteroid/gutlunch/regenerate_icons(new_udder_volume, max_udder_volume)
 	cut_overlays()
-	if(udder.reagents.total_volume == udder.reagents.maximum_volume)
-		add_overlay("gl_full")
+	var/static/gutlunch_full_overlay = iconstate2appearance(icon, "gl_full")
+	if(new_udder_volume == max_udder_volume)
+		add_overlay(gutlunch_full_overlay)
```

Comment:
changed this overlay to use iconstate2appearance as per contribution guidelines but now it fails integration tests... paging @MrStonedOne

Reply:
For whatever reason, `icon` is just null.

Try deferring the initialization of the static to when its first used. 

As in:

```
var/static/gutlunch_full_overlay
if (isnull(gutlunch_full_overlay))
    gutlunch_full_overlay = iconstate2appearance(icon, "gl_full")
```

Reply:
thats annoying.

lummox told me statics initialized on first use.

It seems they initialize with global variables, meaning icon is null because src is null.

initial(icon) could produce useful results too.